rcljava/src/main/java/org/ros2/rcljava/qos/QoSProfile.java

```diff
@@ -17,37 +17,56 @@
 
 import java.time.Duration;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.ros2.rcljava.common.JNIUtils;
 import org.ros2.rcljava.qos.policies.Durability;
 import org.ros2.rcljava.qos.policies.History;
 import org.ros2.rcljava.qos.policies.Liveliness;
 import org.ros2.rcljava.qos.policies.QoSPolicy;
 import org.ros2.rcljava.qos.policies.Reliability;
 
+/**
+ * Implementation of the ROS qos profile abstraction.
+ *
+ * It works as a bridge with rmw_qos_profile_t.
+ * This class provides several static methods to instantiate the default profiles,
+ * getters for the different policies, and setters that allow method chaining.
+ *
+ * Examples:
+ *
+ *    QoSProfile.keepLast(10);  // default qos profile with depth 10
+ *    // keep all qos profile with best effort reliability and transient local durability
+ *    QoSProfile.keepAll().
+ *      setReliability(Reliability.BEST_EFFORT).setDurability(Durability.TRANSIENT_LOCAL);
+ *    QoSProfile.sensorData();  // get predefined sensor data qos profile
+ */
 public class QoSProfile {
   // TODO(ivanpauno): Update all qos policies in a way that the objects are created from RCL,
```

Comment:
There were two TODOs:
- Avoiding to hard code the default profiles, that is being fixed here.
- Avoiding to hard code the policies enumeration numbers, that's not being fixed here.

This comment was referring to the second, if it's not clear I can try to rephrase it.

Reply:
Do you think it's worth doing the second point? I don't see an issue using separate enum values within rcljava, though I found a [similar TODO in rclpy](https://github.com/ros2/rclpy/blob/a37c01511324c1e3e57bc47a7e1286bf79514c60/rclpy/rclpy/qos.py#L30).

Reply:
> Do you think it's worth doing the second point at all? I don't see an issue using separate enum values within rcljava, though I found a similar TODO in rclpy.

The issue is that you have to keep the number synced.
If the numbers get out of sync it might be a pretty hard bug to find, causing unexpected behavior to the user.